Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -43,19 +43,46 @@ ScrollAnchoringController::ScrollAnchoringController(ScrollableArea& owningScrol
     : m_owningScrollableArea(owningScroller)
 { }
 
+ScrollAnchoringController::~ScrollAnchoringController()
+{
+    invalidateAnchorElement();
```

Comment:
Removed the setInScrollAnchoringAncestorChain call but I think we should still invalidate the current anchor so that the set of scrollable areas needing update on the FrameView is correct.